File: aws-lambda-functions/python/DynamoDB_Read.py
import boto3
import json

dynamodb = boto3.resource('dynamodb', region_name='eu-central-1')
table = dynamodb.Table('EvaluationTable')

def lambda_handler(event, context):
    unique_id = "cf19b76e-54e6-4672-bcc3-e25bde8721f8"
    try:
        table.get_item(
            Key={
                'Id': unique_id
            }
        )
        
        return {
            'statusCode': 200,
            'body': json.dumps({
                'message': 'Success',
                'id': unique_id
            })
        }
        
    except Exception as e:
        print(f"Fehler beim Lesen in DynamoDB: {e}")
        
        return {
            'statusCode': 500,
            'body': json.dumps({
                'error': 'Interner Serverfehler'
            })
        }

File: aws-lambda-functions/nodejs/DynamoDB_Read.mjs
import { DynamoDBClient, GetItemCommand } from "@aws-sdk/client-dynamodb";

const client = new DynamoDBClient({ region: "eu-central-1" });

export const handler = async () => {
  const id = "cf19b76e-54e6-4672-bcc3-e25bde8721f8"

  try {
    await client.send(new GetItemCommand({
      TableName: "EvaluationTable",
        Key: {"Id": {S: id}}
    }));

    return { 
      statusCode: 200,
      body: JSON.stringify({ message: "Success", id }) 
    };
  } catch (error) {
    console.error("Fehler beim Schreiben in DynamoDB:", error);
    
    return { 
      statusCode: 500,
      body: JSON.stringify({ error: "Interner Serverfehler" }) 
    };
  }
};